Tests/Realm.Tests/Database/RealmValueTests.cs

```diff
@@ -408,6 +408,61 @@ private void RunNumericTests(RealmValue rv, long value, bool isManaged)
             Assert.That(rv != RealmValue.Null);
         }
 
+        [Test]
+        public void EmbeddedObjectAsRealmValue_WhenAddedToRealm_Throws()
+        {
+            _realm.Write(() =>
+            {
+                var rvo = new RealmValueObject
+                {
+                    RealmValueProperty = new EmbeddedIntPropertyObject()
+                };
+                Assert.Throws<NotSupportedException>(() => _realm.Add(rvo));
+
+                var rvoWithList = new RealmValueObject
+                {
+                    RealmValueList = { new EmbeddedIntPropertyObject() }
+                };
+                Assert.Throws<NotSupportedException>(() => _realm.Add(rvoWithList));
+
+                var rvoWithSet = new RealmValueObject
+                {
+                    RealmValueSet = { new EmbeddedIntPropertyObject() }
+                };
+                Assert.Throws<NotSupportedException>(() => _realm.Add(rvoWithSet));
+
+                var rvoWithDictionary = new RealmValueObject
+                {
+                    RealmValueDictionary = { { "embedded", new EmbeddedIntPropertyObject() } }
+                };
+                Assert.Throws<NotSupportedException>(() => _realm.Add(rvoWithDictionary));
+            });
+        }
+
+        [Test]
+        public void EmbeddedObjectAsRealmValue_WhenModifiedInRealm_Throws()
```

Comment:
Similar comment about dividing per code path here.
The team needs to give opinions.